kubernetes/hypertrace.sh

```diff
@@ -92,6 +102,26 @@ fi
 subcommand=$1; shift
 
 case "$subcommand" in
+  dry-run)
+    EXIT_CODE=0;
+    cleanup
+    echo "[INFO] creating helm deployment template for hypertrace data services."
+    helm dependency update ${HYPERTRACE_HOME}/data-services ${HELM_FLAGS}
+    if [ $HT_DATA_STORE == "postgres" ]; then
+      helm template hypertrace-data-services ${HYPERTRACE_HOME}/data-services -f ${HYPERTRACE_HOME}/data-services/values.yaml -f ${HYPERTRACE_HOME}/data-services/postgres/values.yaml -f ${HYPERTRACE_HOME}/clusters/$HT_PROFILE/values.yaml --set htEnv=${HT_ENV} > ${HYPERTRACE_HOME}/data-services/helm-deployment-templates/manifests.yaml
```

Comment:
`helm upgrade` command also accepts `--dry-run` param and the output is similar (rendered templates).
I think it might be worth inspecting the difference between output of these two commands, I feel helm upgrade with dry-run would be closer to the final deployment (since that is actual command used in deployment)

Reply:
actually `helm upgrade` needs deployed release to generate these templates where as `helm template` command can do it even without deploying Hypertrace which is more helpful in our case, I feel.

Reply:
> helm upgrade needs deployed release to generate these templates 

Didn't get your point? What do you mean by helm upgrade need deployed release, with `--dry-run` I guess it only needs the template and values files, yes it also the dependencies to be pulled first, but I guess `template` also needs it

Reply:
Okay. so in ideal world there are two ways to do this with helm without deployment, `helm install --dry-run` or `helm template`. 

`helm upgrade --dry-run` requires connection to the server/ k8s and deployed release because it's **upgrade**. see below image:
<img width="1101" alt="Screenshot 2021-02-11 at 7 48 05 PM" src="https://user-images.githubusercontent.com/26570044/107649712-bf4e9780-6ca3-11eb-8586-bd62c12bb14e.png">

On the other hand, helm template works out of the box without need of existing deployment. if you want alternative to `helm template` then `helm install --dry-run` is proper alternative though if someone is using helm version other than helm 3 this `helm install` command will need connection to tiller which works inside k8s. 

So considering all these things, helm template is safest option which satisfies our use-case and works.

Reply:
Ah right, I was talking referring to `helm install --dry-run` all the while, mixed install and upgrade.

> helm install command will need connection to tiller which works inside k8s.

Makes sense to go with Template in that case.